src/Bicep.Core/Intermediate/ExpressionRewriteVisitor.cs

```diff
@@ -11,14 +11,27 @@ namespace Bicep.Core.Intermediate;
 
 public abstract class ExpressionRewriteVisitor : IExpressionVisitor
 {
+    void IExpressionVisitor.VisitAccessChainExpression(AccessChainExpression expression) => ReplaceCurrent(expression, ReplaceAccessChainExpression);
+    public virtual Expression ReplaceAccessChainExpression(AccessChainExpression expression)
+    {
+        var hasChanges = TryRewriteStrict(expression.FirstLink, out var firstLink);
```

Comment:
Can be rewritten as:
```csharp
var hasChanges =
    TryRewriteStrict(expression.FirstLink, out var firstLink) |
    TryRewriteStrict(expression.AdditionalProperties, out var additionalProperties);

return hasChanges ? new(expression.SourceSyntax, firstLink, additionalProperties) : expression;
```

I've tried to keep a strict structure here because I think we should be able to autogen this file (+ some of the other visitors) in the future.